### PR TITLE
feat: adds hasPendingMappings

### LINKED ISSUE #803: [4.x] Add has_pending_mapping
Please add the following method to `search_client`

```java
function has_pending_mappings(retrieve_mappings: boolean = false, opts: request_options) 
return has_pending_mappings_responses

struct has_pending_mappings_response {
	pending: boolean,
	clusters: map<string, string[]>
}
```

```java
{
  "method": "GET",
  "url": "/1/clusters/mapping/pending?getClusters={retrieve_mappings}",
  "hostType": "read"
}
```

### PR DESCRIPTION
**Pull request by nunomaduro**
This pull requests closes #803.

Requires #1023 needs to be merged first.

### CODE REVIEW

**Comment by Haroenv** [comment]
can you change the base of the PR to be #1023 instead, so the diff is smaller? (stacked PR)

**Comment by nunomaduro** [comment]
@Haroenv Good idea, done!

File: packages/client-search/src/types/HasPendingMappingsOptions.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export type HasPendingMappingsOptions = {
+  /**
+   * If the clusters pending mapping state should be on the response.
+   *
+   * @alias getClusters
```
Comment (Haroenv) [inline packages/client-search/src/types/HasPendingMappingsOptions.ts]: if this is an alias, why do we support both?

File: packages/client-search/src/methods/client/hasPendingMappings.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export const hasPendingMappings = (base: SearchClient) => {
+  return (
+    requestOptions?: HasPendingMappingsOptions & RequestOptions
+  ): Readonly<Promise<HasPendingMappingsResponse>> => {
+    const { retrieveMappings, ...options } = requestOptions || {};
+
+    if (retrieveMappings === true) {
+      // eslint-disable-next-line functional/immutable-data
+      options.getClusters = true;
```
Comment (Haroenv) [inline packages/client-search/src/methods/client/hasPendingMappings.ts]: I don't see any unit tests for this, on purpose?

Comment (nunomaduro) [reply packages/client-search/src/methods/client/hasPendingMappings.ts]:
```
  await expect(client.hasPendingMappings({ retrieveMappings: true })).resolves.toMatchObject({
     clusters: expect.any(Object),
     pending: expect.any(Boolean),
   });
```

File: packages/client-search/src/types/HasPendingMappingsOptions.ts (hunk 1 above)
Comment (nunomaduro) [reply packages/client-search/src/types/HasPendingMappingsOptions.ts]: Actually, I was thinking about removing `alias` here, and marking `getClusters` as internal. WDYT?

File: packages/client-search/src/methods/client/hasPendingMappings.ts (hunk 2 above)
Comment (Haroenv) [reply packages/client-search/src/methods/client/hasPendingMappings.ts]: that's an integration test, no?

File: packages/client-search/src/types/HasPendingMappingsOptions.ts (hunk 1 above)
Comment (Haroenv) [reply packages/client-search/src/types/HasPendingMappingsOptions.ts]: if they're just aliases, I'd even argue for not making getClusters an argument

File: packages/client-search/src/methods/client/hasPendingMappings.ts (hunk 2 above)
Comment (nunomaduro) [reply packages/client-search/src/methods/client/hasPendingMappings.ts]: @Haroenv Indeed, but what is the point of having a unit test if the integration one already covers all the cases?

File: packages/client-search/src/types/HasPendingMappingsOptions.ts (hunk 1 above)
Comment (nunomaduro) [reply packages/client-search/src/types/HasPendingMappingsOptions.ts]: I will add getClusters  it as `@internal` so we have type here: https://github.com/algolia/algoliasearch-client-javascript/blob/858e43dfacef7d26d1a44b147741183b0e8677b3/packages/client-search/src/methods/client/hasPendingMappings.ts#L14

**Pull request merged**